lib/assets/javascripts/twine.js.coffee

```diff
@@ -268,10 +268,19 @@ Twine.bindingTypes =
   'bind-class': (node, context, definition) ->
     fn = wrapFunctionString(definition, '$context,$root', node)
     lastValue = {}
+    additions = []
+    removals = []
     return refresh: ->
       newValue = fn.call(node, context, rootContext)
-      for key, value of newValue when !lastValue[key] != !value
-        $(node).toggleClass(key, !!value)
+      for key, value of newValue
+        value = newValue[key] = !!newValue[key]
+        cached = lastValue[key]
+        if cached == undefined and $(node).hasClass(key) != value or cached != value
```

Comment:
I would consider replacing the `cached == undefined and $(node).hasClass(key) != value` part with `cached ?= $node.hasClass(key)` on the previous line.